packages/roosterjs-editor-dom/lib/selection/experimentalAttributeBasedSelectionMarker.ts

```diff
@@ -0,0 +1,137 @@
+import Position from './Position';
+import createRange from './createRange';
+import queryElements from '../utils/queryElements';
+
+const SELECTION_START_OFFSET_ATTRIBUTE = 'data-offset1';
+const SELECTION_END_OFFSET_ATTRIBUTE = 'data-offset2';
```

Comment:
these could probably be made more complicated to avoid collisions with native DOM content